Automattic/woocommerce-payments
# Issue/5697 add deposit business day delay notice

Fixes #5697 

#### Changes proposed in this Pull Request

This PR utilises the changes in `issue/5798-add-banner-notice` (PR #5814) to add a notice to the Deposits signalling that deposits may take 1-2 business days. 


<p align="center">
<img width="697" alt="Screenshot 2023-04-11 at 2 35 17 pm" src="https://user-images.githubusercontent.com/8490476/231057293-1918246c-4a22-4e6c-8401-3a65c12c73ec.png">
</p>

#### Testing instructions

1. Checkout this branch.
2. Enable the deposits overview updates by flipping on the feature flag.
   - `update_option( '_wcpay_feature_simplify_deposits_ui', 1 );`
3. Go to the **Payments > Overview** admin page.
4. Depending on your store you will either see one of the following results. 

| 1 Pending        | 1 In-transit           | All Sent | 
|:-------------:|:-------------:|:-----:|
| <img width="697" alt="Screenshot 2023-04-11 at 2 35 17 pm" src="https://user-images.githubusercontent.com/8490476/231057293-1918246c-4a22-4e6c-8401-3a65c12c73ec.png"> | <img width="701" alt="Screenshot 2023-04-11 at 2 36 27 pm" src="https://user-images.githubusercontent.com/8490476/231057447-c4eb41ec-7d3f-49ff-9273-d802c6afba02.png"> | <img width="699" alt="Screenshot 2023-04-11 at 2 37 42 pm" src="https://user-images.githubusercontent.com/8490476/231057553-6b9d4e91-70cf-40ad-81bf-e35f762414d7.png"> | 




| Multiple pending/in-transit        | Only 1 In-transit | No deposit |
|:-------------:|:-------------:|:-----:|
|  <img width="699" alt="Screenshot 2023-04-11 at 2 31 32 pm" src="https://user-images.githubusercontent.com/8490476/231056709-599b028e-5e7f-4cb2-9017-3635401ed580.png"> |  <img width="696" alt="Screenshot 2023-04-11 at 2 30 27 pm" src="https://user-images.githubusercontent.com/8490476/231056549-77b094d5-0ebd-4e9c-bf60-d4bcec933b99.png">| <img width="699" alt="Screenshot 2023-04-11 at 2 39 00 pm" src="https://user-images.githubusercontent.com/8490476/231057715-7ed3819b-75f9-4aae-83ab-188ca160471e.png">   |




To test the various statuses I had to mock the response from `useRecentDeposits`. For example, here's a code snippet of what I did locally to easily test the various deposit statuses without needing to wait for Stripe to generate a deposit. 

```php
const RecentDepositsList: React.FC< RecentDepositsProps > = ( {
	currency,
}: RecentDepositsProps ): JSX.Element => {
	//const recentDeposits = useRecentDeposits( currency );
	const recentDeposits = {
		isLoading: false,
		deposits: [
			{
				id: '1',
				date: '2023-04-10',
				type: 'deposit',
				amount: 2982,
				currency: 'USD',
				fee_percentage: 0,
				fee: 0,
				status: 'in_transit',
				bankAccount: '123456789',
				automatic: true,
			} as CachedDeposit,
			{
				id: '2',
				date: '2023-03-09',
				type: 'deposit',
				amount: 13721,
				currency: 'USD',
				fee_percentage: 0,
				fee: 0,
				status: 'paid',
				bankAccount: '987654321',
				automatic: true,
			} as CachedDeposit,
			{
				id: '2',
				date: '2023-02-09',
				type: 'deposit',
				amount: 12314,
				currency: 'USD',
				fee_percentage: 0,
				fee: 0,
				status: 'paid',
				bankAccount: '987654321',
				automatic: true,
			} as CachedDeposit,
		],
	};
	const isLoading = recentDeposits.isLoading;
...
```




-------------------


**Post merge**
